ev-dev-tools/src/ev_cli/templates/ld-ev.cpp.j2

```diff
@@ -84,8 +85,17 @@ std::vector<Everest::cmd> everest_register() {
     adapter.gather_cmds(*p_{{ impl.id }});
 
     {% endfor %}
-    {% for interface in requires %}
-    auto r_{{ interface.id }} = std::make_unique<{{ interface.class_name }}>(&adapter, "{{ 'optional:' if interface.optional }}{{ interface.id }}");
+    {% for requirement in requires %}
+    {% if requirement.is_vector %}
+    auto r_{{ requirement.id }} = std::make_unique<std::vector<std::shared_ptr<{{ requirement.class_name }}>>>();
+    if (connections.contains("{{ requirement.id }}")) {
+        for (uint64_t idx = 0; idx < connections.size(); idx++) {
+            r_{{ requirement.id }}->push_back(std::make_shared<{{ requirement.class_name }}>(&adapter, "{{ requirement.id }}", idx));
```

Comment:
I don't really see the point of using std::shared_ptr here, couldn't we just drop that?

Reply:
we could, yes